monai/networks/nets/hovernet.py

```diff
@@ -391,6 +391,19 @@ class Mode(Enum):
         FAST: int = 0
         ORIGINAL: int = 1
 
+    class Branch(StrEnum):
+        """
+        Three branches of HoVerNet model, which results in three outputs:
+        `HOVER` is horizontal and vertical regressed gradient map of each nucleus,
+        `NUCLEUS` is the segmentation of all nuclei, and
+        `TYPE` is the type of each nucleus.
+
+        """
+
+        HOVER = "horizontal_vertical"
+        NUCLEUS = "nucleus_prediction"
+        TYPE = "type_prediction"
```

Comment:
I would suggest the name to be consistent with the paper which users can easily find assistance from the paper.
```
HOVER = "horizontal_vertical"
NP = "nuclear_pixel"
NC = "nuclear_classification"
```

Reply:
I like the idea of referring to the original paper but the reason for using keys, rather than just numeric indices, is to make clear what each branch does. NC, NP and HOVER are not very clear in their meaning, whereas the the text is. If we are going to use keys then I think drBeh's are clearer.

Reply:
Thanks @KumoLiu and @JHancox for your comments. 
In my opinion, we should either be descriptive or minimalistic, so I would suggest to take the following keys, which are easier to use, have the same structure (abbreviation of the descriptive names), and mostly matches the paper. What do you think?
```python
HV = "horizontal_vertical"
NP = "nuclear_pixel"
NC = "nuclear_classification"
```

Reply:
For the description to use, for instance "type_prediction" or "nuclear_classification", I would rely on @JHancox's opinion since he has more experience with this model.

Reply:
I agree with the change of **key** and will be willing to listen to @JHancox's opinion on the description!

Reply:
This information is represented as a class instance of `HoVerNet` model, and not being reflected in MetaTensor. MetaTensor carries information about the image. Anyways, the end user does not need to deal with the branch names and needs to work only with high-level API but developers can access the branch names to use it in other related components like for testing:
https://github.com/Project-MONAI/MONAI/blob/67cee72eaf68f5b13bfa0326eb8253e00d4cab22/tests/test_hovernet.py#L51